### PR TITLE
to属性でオブジェクトを使用するように変更

### LINKED ISSUE #332: backボタン押すと404
https://q.trap.jp/messages/76d36bfa-16f7-499a-9d07-4c3539787971

### PR DESCRIPTION
**Pull request by dye8128**
### **User description**
close #332
#314 の対応(to属性にstring型変数が指定されているものについては未完)

___

### **PR Type**
Enhancement

___

### **Description**
- `router-link`の`to`属性を文字列からオブジェクト形式に変更
- 各コンポーネントおよびページでのルーティングの一貫性を向上
- コードの可読性と保守性を向上

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>22 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>ContestTeamItem.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Contest/ContestTeamItem.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-87c5ef6ed2277353847e51f5c453dc4ad45a054c328484514f82386d1c348949">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestTeams.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Contest/ContestTeams.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-715a5b5197b6a9ff42072913e621033f3134af98c2a20c6ab3a9252699e60956">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestItem.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Contests/ContestItem.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-390d39f21fca2a057a0094b89134676096bba43429f4f75b34e148dfb29e8f5d">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>EventItem.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Events/EventItem.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-aa5185b28db246dcf9ddda99daf19c885b5e97e24c4b18033505094d0f65f3ae">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>PageHeader.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Layout/PageHeader.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-e6d0ef9650f7bec3fcc659f9af98c52fb591652e0a6429b46d160a588ec97f21">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ProjectItem.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Projects/ProjectItem.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-844a39636e9a996623141b9ea32115ddb5552c6beeed673366b54fa43ae93dbd">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserProfileDesktop.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/User/UserProfileDesktop.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-00724c933f562b5a69e6898067602d1a5a58938f76eafea4463c20da4440b09a">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserProfileMobile.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/User/UserProfileMobile.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-73f870cbca3afd6225dd183c9e6139f0255449651aee312b3e6a2d3375a5ab8b">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>AccountItem.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/UserAccounts/AccountItem.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-80a079fa83e99b29e618528a6dd3f03447bdfaed58415e39240d7fcb39157cef">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Contest.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Contest.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-d558762d359ad5f4ae0424cef5d1ff91c43f5e01633390e791dc37346b1244a6">+5/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestEdit.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestEdit.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-6b345b9b3a019b15846536d31ccad1be7489f3dcc49e2f18453e8701556194be">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestNew.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestNew.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-2480a3520085ba035bf91cec23e66691c4bb8398c16ea8785b978e50a72ba04e">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestTeamEdit.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestTeamEdit.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-fba1e6a0cebaa7278ce05dc4325ed0db0cdafa561086a629740d23d3ae69d2b3">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ContestTeamNew.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ContestTeamNew.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-0cf527c2206797a81333cbb51d2070441144d1dea8944380cb84cb4516818a10">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Contests.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Contests.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-00aadddd7146b1fe74e665c6f9e4354eb37da680aa61978bd6d579d54dd70605">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Event.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Event.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-a575e96bf56c7266c57cdd22bc6606d206114e8988880254045b548acdaa9de0">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Project.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Project.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-4e8aa41e439d638b3e4fb2ed6b5fc5fd543a6e78b8e5431f82746141b203122a">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ProjectNew.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/ProjectNew.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-02e0c08d10c3b4738460539a64f12e91bdfaaec3367a440492d3e31a4946e5da">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Projects.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/Projects.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-363eca605e87df7acfca10ba0acb91ccc250494d61cfbb3c58fdafbc6ca9a82f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccountEdit.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccountEdit.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-6dd39e0ccbeb6480292f207d72db2c009a37026942f2c4a8082d94037f5e2494">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccountNew.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccountNew.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-0e1077c6ce1d02539bd752d0b681d55167c2d645ce8deaf4ceae65eb1ba65de9">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UserAccounts.vue</strong><dd><code>`router-link`の`to`属性をオブジェクト形式に変更</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/pages/UserAccounts.vue

- `router-link`の`to`属性をオブジェクト形式に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/traPortfolio-Dashboard/pull/334/files#diff-8fa659e3153dfa9e05cfc9887d7bce3d6fb7562066f065335a501f193c5e4c69">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by Pugma** [comment]
formatだけ忘れてそうなのでお願いします！

**Comment by dye8128** [comment]
formatかけたので確認お願いします！

File: src/pages/Project.vue [hunk 1]
```diff
@@ -160,7 +160,7 @@ const handleDelete = (id: string) => {
     </form>
 
     <div :class="$style.buttonContainer">
-      <router-link :to="`/projects/${projectId}`" :class="$style.link">
+      <router-link :to="{ name: 'Projects' }" :class="$style.link">
```
Comment (ogu-kazemiya) [inline src/pages/Project.vue:163]: 対応するnameは`'Project'`だと思います

**Review by ogu-kazemiya — COMMENTED** [review]
他のところは問題なさそうです！

File: src/pages/Project.vue (hunk 1 above)
Comment (dye8128) [reply src/pages/Project.vue:163]: そこはcloseに指定しているIssueの変更点です
(`/projects/${ProjectId}`では404が返ってくるというもの)
なのでこのままで大丈夫です！
Pina君に確認すれば分かるかと

**Review by mehm8128 — COMMENTED** [review]
https://github.com/traPtitech/traPortfolio-Dashboard/issues/314
もcloseに指定しておくとよさそうです(メインはこれの対応だと思うので))

**Comment by dye8128** [comment]
確かに #314 への対応が主なのですが、string型変数が :to に指定されている箇所があり(PageInfoPanel.vueなど)、これの対応が終わっていないためcloseに指定していないです
これの対応については、別ブランチを立てて作業中です(fix/url-type)

そのため一時的な対応として、 #332 のcloseと #334 の一部を含むPRを出しました
(Pugma君とのペアプロでこの対応をすることになりました)

**Comment by mehm8128** [comment]
なるほどです
ではcloseまで書かなくても、関連issueとしてPRの説明文にissueのURLを貼っておく(`close`とか書かずにURL貼るだけなら元issueはcloseされないので)と、後で見返すときになんでこのPRが作られたのか等が分かりやすくなりそうです

**Comment by dye8128** [comment]
分かりました、ありがとうございます！
PR文を訂正しました！

**Review by ogu-kazemiya — APPROVED** [review]
OKです

**Pull request merged**